### PR TITLE
[SPARK-20396][SQL][PySpark][FOLLOW-UP] groupby().apply() with pandas udf

### PR DESCRIPTION
**Pull request by ueshin**
## What changes were proposed in this pull request?

This is a follow-up of #18732.
This pr modifies `GroupedData.apply()` method to convert pandas udf to grouped udf implicitly.

## How was this patch tested?

Exisiting tests.

### CODE REVIEW

File: python/pyspark/sql/functions.py [hunk 1]
```diff
@@ -2038,13 +2038,22 @@ def _wrap_function(sc, func, returnType):
                                   sc.pythonVer, broadcast_vars, sc._javaAccumulator)
 
 
+class PythonUdfType(object):
+    # row-based UDFs
+    NORMAL_UDF = 0
+    # single-row vectorized UDFs
```
Comment (cloud-fan) [inline python/pyspark/sql/functions.py]: one-to-one mapping vectorized UDFs

**Comment by gatorsmile** [comment]
retest this please

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -2038,13 +2038,22 @@ def _wrap_function(sc, func, returnType):
                                   sc.pythonVer, broadcast_vars, sc._javaAccumulator)
 
 
+class PythonUdfType(object):
+    # row-based UDFs
```
Comment (gatorsmile) [inline python/pyspark/sql/functions.py]: Nit: Please update all `row-based UDFs` to `row-at-a-time UDFs`

File: sql/core/src/main/scala/org/apache/spark/sql/execution/python/UserDefinedPythonFunction.scala [hunk 3]
```diff
@@ -22,17 +22,26 @@ import org.apache.spark.sql.Column
 import org.apache.spark.sql.catalyst.expressions.Expression
 import org.apache.spark.sql.types.DataType
 
+private[spark] object PythonUdfType {
+  // row-based UDFs
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/python/UserDefinedPythonFunction.scala]: The same here.

**Comment by gatorsmile** [comment]
LGTM. 

@ueshin Could you remove `[WIP]` from the title of this PR?

File: python/pyspark/sql/functions.py (hunk 2 above)
Comment (ueshin) [reply python/pyspark/sql/functions.py]: Sure, I'll update it.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/python/UserDefinedPythonFunction.scala (hunk 3 above)
Comment (ueshin) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/python/UserDefinedPythonFunction.scala]: Sure, I'll update it, too.

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
retest this please

**Pull request closed without merging**